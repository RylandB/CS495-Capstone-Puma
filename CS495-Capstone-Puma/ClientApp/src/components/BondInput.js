import React from 'react'

let state = {assets: [
        { name: 'test1', price: '45.5', quantity: 21, dateOfIssue: "04/30/1997", dateOfMaturity: "01/01/2030",
            
            incomePaymentMonth: 12345, incomePaymentDay: 54321, accrualMethodType: "annually",
            
            callDate: "12/12/2012", callPrice: "01/23/1245", dateOfFirstPayment: "11/25/2005"}
    ],
    inputName: null,
    inputPrice: null,
    inputQuantity: null,
    inputDateOfIssue: null,
    inputDateOfMaturity: null,
    inputIncomePaymentMonth: null,
    inputIncomePaymentDay: null,
    inputAccrualMethodType: null,
    inputCallDate: null,
    inputCallPrice: null,
    inputDateOfFirstPayment: null
};

export class BondInput extends React.Component{
    
    constructor(props) {
        super(props);
        this.state = state;
        this.addAsset = this.addAsset.bind(this);
        this.removeAsset = this.removeAsset.bind(this);
        this.handleInputChange = this.handleInputChange.bind(this);
    }

    componentWillUnmount() {
        state = this.state;
    }

    renderTableHeader() {
        let header = Object.keys(this.state.assets[0]);
        return header.map((key, index) => {
            return <th key={index}>{key.toUpperCase()}</th>
        })
    }

    renderTableData() {
        return this.state.assets.map((asset, index) => {
            const { name, price, quantity, dateOfIssue,
                
                dateOfMaturity, incomePaymentMonth, incomePaymentDay,
                
                accrualMethodType, callDate, callPrice, dateOfFirstPayment} = asset;
            return (
                <tr key={name}>
                    <td>{name}</td>
                    <td>{price}</td>
                    <td>{quantity}</td>
                    <td>{dateOfIssue}</td>
                    <td>{dateOfMaturity}</td>
                    <td>{incomePaymentMonth}</td>
                    <td>{incomePaymentDay}</td>
                    <td>{accrualMethodType}</td>
                    <td>{callDate}</td>
                    <td>{callPrice}</td>
                    <td>{dateOfFirstPayment}</td>
                </tr>
            )
        })
    }

    handleInputChange(event){

        const target = event.target;
        const value = event.target.value;
        const name = target.name;

        this.setState({
            [name]: value
        });

    }

    async addAsset(){
        const newAsset = {
            name: this.state.inputName,
            price: this.state.inputPrice,
            quantity: this.state.inputQuantity,
            dateOfIssue: this.state.inputDateOfIssue,
            dateOfMaturity: this.state.inputDateOfMaturity,
            incomePaymentMonth: this.state.inputIncomePaymentMonth,
            incomePaymentDay: this.state.inputIncomePaymentDay,
            accrualMethodType: this.state.inputAccrualMethodType,
            callDate: this.state.inputCallDate,
            callPrice: this.state.inputCallPrice,
            dateOfFirstPayment: this.state.inputDateOfFirstPayment
        };
        await this.setState({assets: this.state.assets.concat(newAsset)});
        this.props.bondCallback(this.state.assets);
    }

    async removeAsset(){
        await this.setState({asset: this.state.assets.pop()});
        this.props.bondCallback(this.state.assets);
    }

    render(){
        return(
            <div>
                <h3 id='title'>Assets</h3>
                <table className='table' id='assets'>
                    <tbody>
                    <tr>{this.renderTableHeader()}</tr>
                    {this.renderTableData()}
                    </tbody>
                </table>
                <label>Name</label>
                <input type="text" name="inputName" onChange={this.handleInputChange} value={this.state.inputName}/>
                <br />
                <label>Price</label>
                <input type="number" name="inputPrice" onChange={this.handleInputChange} value={this.state.inputPrice}/>
                <br />
                <label>Quantity</label>
                <input type="number" name="inputQuantity" onChange={this.handleInputChange} value={this.state.inputQuantity}/>
                <br />
                <label>DateOfIssue</label>
                <input type="date" name="inputDateOfIssue" onChange={this.handleInputChange} value={this.state.inputDateOfIssue}/>
                <br />
                <label>DateOfMaturity</label>
                <input type="date" name="inputDateOfMaturity" onChange={this.handleInputChange} value={this.state.inputDateOfMaturity}/>
                <br />
                <label>IncomePaymentMonth</label>
                <input type="number" name="inputIncomePaymentMonth" onChange={this.handleInputChange} value={this.state.inputIncomePaymentMonth}/>
                <br />
                <label>IncomePaymentDay</label>
                <input type="number" name="inputIncomePaymentDay" onChange={this.handleInputChange} value={this.state.inputIncomePaymentDay}/>
                <br />
                <label>AccrualMethodType</label>
                <input type="text" name="inputAccrualMethodType" onChange={this.handleInputChange} value={this.state.inputAccrualMethodType}/>
                <br />
                <label>CallDate</label>
                <input type="date" name="inputCallDate" onChange={this.handleInputChange} value={this.state.inputCallDate}/>
                <br />
                <label>CallPrice</label>
                <input type="date" name="inputCallPrice" onChange={this.handleInputChange} value={this.state.inputCallPrice}/>
                <br />
                <label>DateOfFirstPayment</label>
                <input type="date" name="inputDateOfFirstPayment" onChange={this.handleInputChange} value={this.state.inputDateOfFirstPayment}/>
                <br />
                <button onClick={this.addAsset}>Add Asset</button>
                <button onClick={this.removeAsset}> Remove Asset</button>
                <br/>
                <br/>
            </div>
        );
    }

}