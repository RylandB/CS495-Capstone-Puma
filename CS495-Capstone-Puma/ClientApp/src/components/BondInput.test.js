import {BondInput} from "./BondInput"
import React from "react";
it("Testing Jest", () => {
    expect(1+1).toEqual(2);
});

it("Testing Jest 2", () => {
    expect(2+2===4).toEqual(true);
});

it("BondInput Renders Correctly", () => {
    const wrapper = shallow(
        <BondInput />
    );
    
    expect(wrapper).toMatchSnapshot();
});