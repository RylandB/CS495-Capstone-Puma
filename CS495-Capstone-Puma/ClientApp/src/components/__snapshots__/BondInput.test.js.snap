// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BondInput Renders Correctly 1`] = `
<div>
  <h3
    id="title"
  >
    Bonds
  </h3>
  <table
    className="table"
    id="assets"
  >
    <tbody>
      <tr>
        <th
          key="0"
        >
          NAME
        </th>
        <th
          key="1"
        >
          PRICE
        </th>
        <th
          key="2"
        >
          QUANTITY
        </th>
        <th
          key="3"
        >
          DATEOFISSUE
        </th>
        <th
          key="4"
        >
          DATEOFMATURITY
        </th>
        <th
          key="5"
        >
          INCOMEPAYMENTMONTH
        </th>
        <th
          key="6"
        >
          INCOMEPAYMENTDAY
        </th>
        <th
          key="7"
        >
          ACCRUALMETHODTYPE
        </th>
        <th
          key="8"
        >
          CALLDATE
        </th>
        <th
          key="9"
        >
          CALLPRICE
        </th>
        <th
          key="10"
        >
          DATEOFFIRSTPAYMENT
        </th>
      </tr>
    </tbody>
  </table>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name
    </label>
    <input
      name="inputName"
      onChange={[Function]}
      required={true}
      type="text"
      value={null}
    />
    <br />
    <label>
      Price
    </label>
    <input
      name="inputPrice"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <label>
      Quantity
    </label>
    <input
      name="inputQuantity"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <label>
      DateOfIssue
    </label>
    <input
      name="inputDateOfIssue"
      onChange={[Function]}
      required={true}
      type="date"
      value={null}
    />
    <br />
    <label>
      DateOfMaturity
    </label>
    <input
      name="inputDateOfMaturity"
      onChange={[Function]}
      required={true}
      type="date"
      value={null}
    />
    <br />
    <label>
      IncomePaymentMonth
    </label>
    <input
      name="inputIncomePaymentMonth"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <label>
      IncomePaymentDay
    </label>
    <input
      name="inputIncomePaymentDay"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <label>
      AccrualMethodType
    </label>
    <input
      name="inputAccrualMethodType"
      onChange={[Function]}
      required={true}
      type="text"
      value={null}
    />
    <br />
    <label>
      CallDate
    </label>
    <input
      name="inputCallDate"
      onChange={[Function]}
      required={true}
      type="date"
      value={null}
    />
    <br />
    <label>
      CallPrice
    </label>
    <input
      name="inputCallPrice"
      onChange={[Function]}
      required={true}
      type="date"
      value={null}
    />
    <br />
    <label>
      DateOfFirstPayment
    </label>
    <input
      name="inputDateOfFirstPayment"
      onChange={[Function]}
      required={true}
      type="date"
      value={null}
    />
    <br />
    <input
      type="submit"
      value="Add Asset"
    />
  </form>
  <button
    onClick={[Function]}
  >
     Remove Asset
  </button>
  <br />
  <br />
</div>
`;
